psutil/_pslinux.py

```diff
@@ -708,13 +708,25 @@ def cpu_freq():
         min and max frequencies are not available and are set to None.
         """
         ret = []
-        with open_binary('%s/cpuinfo' % get_procfs_path()) as f:
-            for line in f:
-                if line.lower().startswith(b'cpu mhz'):
-                    key, value = line.split(b'\t:', 1)
-                    ret.append(_common.scpufreq(float(value), 0.0, 0.0))
+        path = '%s/cpuinfo' % get_procfs_path()
+        if os.path.exists(path):
+            try:
+                with open_binary(path) as f:
+                    for line in f:
+                        if line.lower().startswith(b'cpu mhz'):
+                            key, value = line.split(b'\t:', 1)
+                            ret.append(_common.scpufreq(float(value), 0., 0.))
+            except IOError as err:
+                raise NotImplementedError(
+                    "%r for file %r" % (err, path))
```

Comment:
Mmm... Not sure I understand why catching IOError is necessary here. Isn't the function declaration guarded by  `if os.path.exists("/proc/cpuinfo"):` already?

Reply:
It's because of this test: https://github.com/davidbrochart/psutil/blob/fix_cpu_freq/psutil/tests/test_linux.py#L859-L868

Reply:
And this test: https://github.com/davidbrochart/psutil/blob/56fa1df9bccfc837d9354ebbcc84306dbb7b734c/psutil/tests/test_linux.py#L699-L702